src/main/java/com/ibm/appscan/jenkins/plugin/builders/AppScanBuildStep.java

```diff
@@ -268,15 +285,25 @@ public IResultsProvider call() throws AbortException {
 		    	}
 		    	catch(ScannerException | InvalidTargetException | InterruptedException e) {
 		    		throw new AbortException(Messages.error_running_scan(e.getLocalizedMessage()));
-		    		}
+		    	}
 			}
 		});
 
     	provider.setProgress(new StdOutProgress()); //Avoid serialization problem with StreamBuildListener.
     	build.addAction(new ResultsRetriever(build, provider, m_name));
-    	
-		if(m_wait && shouldFailBuild(provider))
-			throw new AbortException(Messages.error_threshold_exceeded());	
+                
+        if (m_wait && m_failBuildNonCompliance){
```

Comment:
Can you remove this if block and fold the logic into the existing shouldFailBuild() method?  Something like:

  private boolean shouldFailBuild(IResultsProvider provider) throws AbortException{
    	if(!m_failBuild && !m_failBuildNonCompliance)
    		       return false;
 		try {
 			List<FailureCondition> failureConditions = m_failureConditions;
			 		
 			if(m_failBuildNonCompliance) {
 				failureConditions = new ArrayList<FailureCondition>();
 				FailureCondition nonCompliantCondition = new FailureCondition("total", 0);
 				failureConditions.add(nonCompliantCondition);
 			}
 	    	return new ResultsInspector(failureConditions, provider).shouldFail();
 	    } catch(NullPointerException e) {
 	    	throw new AbortException(Messages.error_checking_results(provider.getStatus()));
 	    }
 	}

Reply:
Sure .
I wanted to display different messages on the build page based on whether the build is failed
1. because the build contains non compliant issue : throw new AbortException(Messages.error_noncompliant_issues());
2. because of the issues more than threshold  :throw new AbortException(Messages.error_threshold_exceeded());

Can we still do it or should we keep the same message in either case

Reply:
The different messages makes sense.  In the shouldFailBuild() method we could create a String with it's value set based on whether or not we're doing a noncompliance check or a normal check using the existing if block.  We could then throw the AbortException directly from the shouldFailBuild() method based on the result of the ResultsInspector.shouldFail() call.